recipes-app/node-red-gpio/files/0002-mraa-gpio-Pass-settings-from-js-to-html.patch

```diff
@@ -0,0 +1,396 @@
++
++module.exports.getPlatformTypeStr = function() {
++    switch (mraa.getPlatformType()) {
++        case 0: return "Galileo v1";
++        case 1: return "Galileo v2";
++        case 2: return "Edison Fab C";
++        case 3: return "DE3813 Baytrail";
++        case 4: return "Minnow Max";
++        case 5: return "Raspberry Pi";
++        case 6: return "Beaglebone";
++        case 7: return "Banana";
++        case 26: return "IOT2050";
++        default: return "unknown";
++    }
++}
++
++module.exports.getVersion = function() {
++    return mraa.getVersion();
++}
+\ No newline at end of file
+diff --git a/hardware/intel/mraa-gpio-ain.html b/hardware/intel/mraa-gpio-ain.html
+index 634c824fa78b..91aa302bef9f 100644
+--- a/hardware/intel/mraa-gpio-ain.html
++++ b/hardware/intel/mraa-gpio-ain.html
+@@ -16,24 +16,9 @@
+             return this.name||"A"+this.pin;
+         },
+         oneditprepare: function() {
+-            var pinnow = this.pin;
+-            $.getJSON('mraa-gpio/'+this.id,function(data) {
+-                var t = "unknown";
+-                if (data === 0) { t = "Galileo v1"; }
+-                if (data === 1) { t = "Galileo v2"; }
+-                if (data === 2) { t = "Edison Fab C"; }
+-                if (data === 3) { t = "DE3813 Baytrail"; }
+-                if (data === 4) { t = "Minnow Max"; }
+-                if (data === 5) { t = "Raspberry Pi"; }
+-                if (data === 6) { t = "Beaglebone"; }
+-                if (data === 7) { t = "Banana"; }
+-                if (data === 26) { t = "IOT2050"; }
+-                $('#btype').text(t);
+-                $('#node-input-pin').val(pinnow);
+-            });
+-            $.getJSON('mraa-version/'+this.id,function(data) {
+-                $('#ver-tip').text(data);
+-            });
++            $('#node-input-pin').val(this.pin);
++            $('#btype').text(RED.settings.mraaGpioAinBoardType);
++            $('#ver-tip').text(RED.settings.mraaGpioAinMraaVersion);
+         }
+     });
+ </script>
+diff --git a/hardware/intel/mraa-gpio-ain.js b/hardware/intel/mraa-gpio-ain.js
+index 6867db4df6a1..8d3b19bc53e6 100644
+--- a/hardware/intel/mraa-gpio-ain.js
++++ b/hardware/intel/mraa-gpio-ain.js
+@@ -1,6 +1,7 @@
+ 
+ module.exports = function(RED) {
+     var m = require('mraa');
++    var mraaContext = require('./mraa-context');
+ 
+     function gpioAin(n) {
+         RED.nodes.createNode(this, n);
+@@ -28,5 +29,16 @@ module.exports = function(RED) {
+             node.x.close();
+         });
+     }
+-    RED.nodes.registerType("mraa-gpio-ain", gpioAin);
++    RED.nodes.registerType("mraa-gpio-ain", gpioAin, {
++        settings: {
++            mraaGpioAinBoardType: {
++                value: mraaContext.getPlatformTypeStr(),
++                exportable: true
++            },
++            mraaGpioAinMraaVersion: {
++                value: mraaContext.getVersion(),
++                exportable: true
++            }
++        }
```

Comment:
This block seems to repeat as well for ever node type. Can't that be taken from somewhere pre-filled?

Reply:
Ok, you would have to generalize the setting field names, but that should be doable, no?

Reply:
Tried, but according to node-red document:

> If a node attempts to register a setting that does not meet the naming requirements an error will be logged.

And the `naming requirements` are not clear in the document, per my trying, it must be started with the node name...

I was thinking of using a fake (or dummy) node, if it's setting could be retrieved from other nodes, but that is also nasty...

Seems to me a dark side of node-red nodes...